model/src/main/kotlin/Provenance.kt

```diff
@@ -100,9 +100,9 @@ data class Provenance(
             // If "it" has a resolved revision it must be equal to the resolved revision of vcsInfo, otherwise the
             // revision of "it" has to equal either the revision or the resolved revision of vcsInfo.
             it.type == vcsInfo.type && it.url == vcsInfo.url && it.path == vcsInfo.path
-                    && it.resolvedRevision?.let {
+                    && (it.resolvedRevision?.let {
                 vcsInfo.resolvedRevision == it
-            } ?: vcsInfo.resolvedRevision == it.revision || vcsInfo.revision == it.revision
+            } ?: vcsInfo.resolvedRevision == it.revision || vcsInfo.revision == it.revision)
```

Comment:
As I've stumbled more than once over this code, should we take the chance and rewrite the whole expression to the IMO more clear

            it.type == vcsInfo.type && it.url == vcsInfo.url && it.path == vcsInfo.path && (
                    (vcsInfo.resolvedRevision == it.resolvedRevision ?: it.revision) ||
                    vcsInfo.revision == it.revision
            )